extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -429,43 +426,63 @@ export const popupMessageListener = (request: Request) => {
   };
 
   const addCustomNetwork = () => {
-    const { customNetwork } = request;
+    const { networkDetails } = request;
     const savedNetworks = getSavedNetworks();
 
     // Network Name already used
     if (
       savedNetworks.find(
         ({ networkName }: { networkName: string }) =>
-          networkName === customNetwork.networkName,
+          networkName === networkDetails.networkName,
       )
     ) {
       return {
-        error: MANAGE_NETWORK_ERROR,
+        error: "Network name is already in use",
       };
     }
 
-    const { isSwitchSelected, ...networkDetails } = customNetwork;
     const networksList: NetworkDetails[] = [...savedNetworks, networkDetails];
 
     localStorage.setItem(NETWORKS_LIST_ID, JSON.stringify(networksList));
-    if (isSwitchSelected) {
-      localStorage.setItem(NETWORK_ID, JSON.stringify(networkDetails));
-    }
 
     return {
-      networkDetails,
       networksList,
     };
   };
 
-  const changeNetwork = () => {
+  const removeCustomNetwork = () => {
     const { networkName } = request;
 
-    const savedNetworks = JSON.parse(
-      localStorage.getItem(NETWORKS_LIST_ID) ||
-        JSON.stringify(DEFAULT_NETWORKS),
-    ) as NetworkDetails[];
+    const savedNetworks = getSavedNetworks();
+    const networkIndex = savedNetworks.findIndex(
+      ({ networkName: savedNetworkName }) => savedNetworkName === networkName,
+    );
+
+    savedNetworks.splice(networkIndex, 1);
+
+    localStorage.setItem(NETWORKS_LIST_ID, JSON.stringify(savedNetworks));
+
+    return {
+      networksList: savedNetworks,
+    };
+  };
+
+  const editCustomNetwork = () => {
+    const { networkDetails, networkIndex } = request;
```

Comment:
when editing, any field can change, so we can't identify the old version by any of the field. We just replace it by the index of where it falls in the networksList array instead